src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1281,15 +1282,37 @@ void SetContent(UIElement container, DependencyProperty contentProperty)
 
 					TryRepairContentConnection(containerAsContentControl, item);
 
-					// Set the datacontext first, then the binding.
-					// This avoids the inner content to go through a partial content being
-					// the result of the fallback value of the binding set below.
-					SetContent(containerAsContentControl, ContentControl.DataContextProperty);
-
-					if (!containerAsContentControl.IsContainerFromTemplateRoot && containerAsContentControl.GetBindingExpression(ContentControl.ContentProperty) == null)
+					if (this is ListViewBase && ItemsSource is null)
+					{
+						// We have Items directly without generating from ItemsSource.
+						// In this case, the container does not get any DC, but the child of the container will.
+						// As a hacky workaround in order to set the child DC when it gets added, we wait for the
+						// container to load to run a one-time callback that sets the child(ren) DC.
+						containerAsContentControl.SetValue(DataContextProperty, null);
+						SetContent(containerAsContentControl, ContentControl.ContentProperty);
+						TypedEventHandler<FrameworkElement, object> setDC = null;
+						setDC = (sender, _) =>
+						{
+							foreach (var child in sender.GetChildren())
+							{
+								child.SetValue(DataContextProperty, DataContext, DependencyPropertyValuePrecedences.TemplatedParent);
+							}
+							containerAsContentControl.Loading -= setDC;
+						};
+						containerAsContentControl.Loading += setDC;
+					}
+					else
```

Comment:
This looks very hacky to me, but at this point we don't yet have any children (usually just a single ListViewItem), so we need to wait for the child(ren) to appear in order to set the DataContext. The only thing I could think of is waiting for `Loading`. @jeromelaban

Reply:
@ramezgerges Indeed, this probably not the best way to go. We'll need to revisit this one. Do you know if this is an issue that happens to ListView or ItemsControl are used directly ?

Reply:
If I remember correctly, I was testing with ComboBox and ListView (both ItemsControls) and fixing in ItemsControl directly matched the behaviour in both. So I would say that the problem is most probably at the ItemsControl level.